plugins/vpc-bridge/config/k8s/pipe.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+//go:build enablek8sconnector && windows
+
+package k8s
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	winio "github.com/Microsoft/go-winio"
+	log "github.com/cihub/seelog"
+	"golang.org/x/sys/windows"
+)
+
+const (
+	//PipeConnectionTimeout Represents timeout for reading from connection object.
+	PipeConnectionTimeout = 1 * time.Minute
+
+	//NamedPipePathPrefix Represents prefix for named pipe path.
+	NamedPipePathPrefix = `\\.\pipe\ProtectedPrefix\Administrators\aws-vpc-k8s-connector`
+)
+
+// pipeReadResult Represents result received from named pipe.
+type pipeReadResult struct {
+	output string // Represents message with output data.
+	error  error  // Represents error while getting output.
+}
+
+// newPipe Create pipe object with random named pipe path.
+func newPipe() (string, error) {
+	// Generate GUID random string.
+	g, err := windows.GenerateGUID()
+	if err != nil {
+		return "", fmt.Errorf("error creating unique name for named pipe: %w", err)
+	}
+	// Create pipe path with prefix.
+	pipeName := fmt.Sprintf("%s-%s", NamedPipePathPrefix, g.String())
+	return pipeName, nil
+}
+
+// readResultFromPipe executes in separate go routine. It creates listener over named pipe.
+// Start accepting connections on named pipe and copy output from conn to result channel.
+// It closes result channel, pipe listener, conn.
+// Return error or output over result channel.
+func readResultFromPipe(ctx context.Context, pipeName string, result chan pipeReadResult) {
+	defer close(result) // Defer closing channel from sender.
+
+	// Create pipe and get listener.
+	// listener is closed in main routine.
```

Comment:
No, it is closed right here deferred in line 71.

Reply:
Addressed, deleted wrong comment.